src/lf_msqueue.ml

```diff
@@ -34,84 +36,82 @@ end
 
 module M : S = struct
 
-  module Cas = Kcas.W1
-
   type 'a node =
     | Nil
-    | Next of 'a * 'a node Cas.ref
+    | Next of 'a * 'a node Atomic.t
 
   type 'a t =
-    { head : 'a node Cas.ref ;
-      tail : 'a node Cas.ref }
+    { head : 'a node Atomic.t ;
+      tail : 'a node Atomic.t }
 
   let create () =
-    let head = (Next (Obj.magic (), Cas.ref Nil)) in
-    { head = Cas.ref head ; tail = Cas.ref head }
+    let head = (Next (Obj.magic (), Atomic.make Nil)) in
+    { head = Atomic.make head ; tail = Atomic.make head }
 
   let is_empty q =
-    match Cas.get q.head with
-    | Nil -> failwith "MSQueue.is_empty: impossible"
+    match Atomic.get q.head with
+    | Nil -> raise Empty
```

Comment:
I think the `failwith` is the correct behavior here. These cases cannot happen, and must be a fatal error. I'd revert this and other similar changes.

Reply:
Reverted these changes.